strawberry/subscriptions/protocols/graphql_transport_ws/handlers.py

```diff
@@ -276,10 +276,10 @@ async def operation_task(
             # cleanup in case of something really unexpected
             # wait for generator to be closed to ensure that any existing
             # 'finally' statement is called
-            result_source = self.subscriptions[operation.id]
-            with suppress(RuntimeError):
-                await result_source.aclose()
             if operation.id in self.subscriptions:
+                result_source = self.subscriptions[operation.id]
+                with suppress(RuntimeError):
+                    await result_source.aclose()
```

Comment:
would it be possible to get a test for this? or is too complicated to replicate?